powerauth-push-client/src/main/java/io/getlime/push/client/PushServerClient.java

```diff
@@ -749,17 +753,17 @@ public Response removeAndroid(Long id) throws PushServerClientException {
      * @throws IOException In case JSON processing fails.
      */
     @SuppressWarnings("unchecked")
-    private <T> T checkHttpStatus(TypeReference<? extends Response> typeReference, HttpResponse<String> response) throws IOException, PushServerClientException {
-        if (response.getStatus() == 200) {
-            return (T) mapper.readValue(response.getBody(), typeReference);
+    private <T> T checkHttpStatus(ParameterizedTypeReference<? extends Response> typeReference, ClientResponse response) throws IOException, PushServerClientException {
+        if (!response.statusCode().isError()) {
```

Comment:
Could we make this condition positive (maybe `is2xxSuccessful()`) or flip the branches? The `!` with `isError()` feels like a "double negation"...